docs/linter.md

```diff
@@ -226,14 +226,15 @@ Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.
 """  # noqa: E501
 ```
 
-To ignore all violations across an entire file, add `# ruff: noqa` to any line in the file, like so:
+To ignore all violations across an entire file, add a `# ruff: noqa` comment on its own line
+anywhere in the file preferably towards the top, like so:
```

Comment:
Perhaps we could say "... add the line `# ruff: noqa` anywhere in the file..." which feels a bit clear than "add a comment on its own line"? Perhaps with another clarifying note at the end of the section saying "Global noqa comments must be on their own line to disambiguate from comments which ignore violations on a single line"?

Reply:
From my admittedly simple tests (as I'm just learning to program) even the `# ruff: noqa` comment must be on its own line. If it is an inline comment then errors are not ignored. See

![image](https://github.com/astral-sh/ruff/assets/13859009/a4a48f61-73c8-4893-a502-65d4ed421917)

(where the error in lines 2 and 4 is `E741 Ambiguous variable name: 'O'`, although I don't understand why it is only highlighted on the 'O' and not the 'X' in line 1)

vs

![image](https://github.com/astral-sh/ruff/assets/13859009/6886cad7-bcdb-4e8a-8dcc-96477e675826)

Reply:
Oh that shouldn't conflict with what I'm suggesting. Ignoring violations on a single line is "add `# noqa` or `# noqa: RULE` to the line" whereas a global violation ignore is adding a new line with `# ruff: noqa`

Reply:
I see. I've pushed a new comment to reword the docs as you suggest. Please let me know if you I should make further changes. Should I squash these changes into one commit and rebase onto the latest commit on `main`?

Reply:
We squash merge so you don't need to do anything with your commit.

I can take a look at that error later today!

Reply:
re the ambiguous name, "I" and "O" are the only names that are considered ambiguous https://docs.astral.sh/ruff/rules/ambiguous-variable-name/

Reply:
Ah, interesting. Thank you for taking the time to look into this.